app/ui.js

```diff
@@ -1718,6 +1720,15 @@ const UI = {
         selectbox.options.add(optn);
     },
 
+    getRSAKey(e) {
+        const key = WebUtil.readSetting("rsa_key");
+        UI.rfb.setRSAKey(key);
+    },
+
+    setRSAKey(e) {
+        WebUtil.writeSetting("rsa_key", e.detail.key);
+    }
```

Comment:
I guess we need some UI for users to be able to provide the key as well.

Reply:
How about generating a new private key for every connection? So we don't need to bother saving the private key securely.

Reply:
That usually takes a very long time. Have you tested it and see how long it takes?

Reply:
I tried to generate a 2048bit rsa key using SubtleCrypto for 100 times and it only took about 6 seconds. So I think on modern browsers it's quite fast.

Reply:
Fair enough. Let's try that then.